rpc/client.go

```diff
@@ -273,26 +273,57 @@ func (c *Client) SetHeader(key, value string) {
 	conn.mu.Unlock()
 }
 
+type callArgs struct {
+	list   []interface{}
+	object interface{}
+}
+
+// CallArgsOpt is created by calling PositionalArgs, NamesArgs or NoArgs
+// and controls the encoding of arguments in the RPC request.
+type CallArgsOpt func(*callArgs)
+
+// PositionalArgs is used to provide args in the server-expected order.
+// They will be encoded as an array.
+func PositionalArgs(args ...interface{}) CallArgsOpt {
+	return func(ca *callArgs) {
+		ca.list = args
+	}
+}
+
+// NamedArgs is used to provide args as a struct or map
+// with member names that match the server-expected arg names.
+// They will be encoded as an object.
+func NamedArgs(arg interface{}) CallArgsOpt {
+	return func(ca *callArgs) {
+		ca.object = arg
+	}
+}
+
+// NoArgs is used to specify that there are no args.
+func NoArgs() CallArgsOpt {
+	return func(ca *callArgs) {}
+}
+
```

Comment:
Here are the new "args options" to choose from. The two funcs below are updated to accept args in this way, and basically the rest of the changes in this PR are just refactoring to accommodate that change.